sstables/sstable_directory.cc

```diff
@@ -73,7 +73,7 @@ sstable_directory::sstable_directory(sstables_manager& manager,
 {}
 
 void sstable_directory::filesystem_components_lister::handle(sstables::entry_descriptor desc, fs::path filename) {
-    if ((desc.generation.as_int() % smp::count) != this_shard_id()) {
+    if (!desc.generation.maybe_owned_by_this_shard()) {
```

Comment:
We can later decorate sstable_directory with some noncopyable_function<shard_id (generation_type)> to communicate how different tables place sstables into shards.

Reply:
added as a TODO in the comment.